### PR TITLE
Allow edge-proxy to connect through http-proxy

### PR DESCRIPTION
**Pull request by kylestein-arm**
Added patch file for edge-proxy to allow configurable external http
proxy

Signed-off-by: Kyle Stein <kyle.stein@arm.com>

### CODE REVIEW

**Comment by costanic** [comment]
`Pulling edge-proxy 
Usage: snapcraftctl pull [OPTIONS]
Try "snapcraftctl pull --help" for help.

Error: Got unexpected extra arguments (git am /var/jenkins_home/workspace/snap-pelion-edge/files/edge-proxy/patches/0001-Add-configurable-HTTP-proxy.patch)
Downloading 'node-v8.12.0-linux-x64.tar.gz'   0%                               
Downloading 'node-v8.12.0-linux-x64.tar.gz' 100%                               
Failed to run 'override-pull': Exit code was 2.`

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -252,6 +252,9 @@ parts:
       source: git@github.com:armPelionEdge/edge-proxy.git
       source-commit: 470539b560a9aea3c8b6ecc0c0e0d3b4064bacaf
       go-importpath: github.com/armPelionEdge/edge-proxy
+      override-pull:
```
Comment (costanic) [inline snap/snapcraft.yaml:255]: missing a pipe '|' at end of this line.

**Comment by kylestein-arm** [comment]
Underlying PR to edge-proxy is being reviewed and needs style changes, so patch file is likely to change

File: snap/snapcraft.yaml (hunk 1 above)
Comment (kylestein-arm) [reply snap/snapcraft.yaml]: fixed

**Comment by kylestein-arm** [comment]
Patch file reflects updated edge-proxy repo PR.   Also added config variable.

**Comment by costanic** [comment]
it looks like the edge-proxy PR was merged, so does that mean we can remove the patch file from this PR?

**Comment by kylestein-arm** [comment]
@costanic, what is our process for bumping the version of edge-proxy in snapcraft.yaml?  It currently points to a sha a few commits behind master, so the patch file is still needed.

**Comment by costanic** [comment]
> @costanic, what is our process for bumping the version of edge-proxy in snapcraft.yaml? It currently points to a sha a few commits behind master, so the patch file is still needed.

@kylestein-arm  you need to bump the SHA in the build recipe for that component.  For edge-proxy, that's here: https://github.com/armPelionEdge/snap-pelion-edge/blob/3978aa3a6c4fe1e53d7faac070b53c5160171a55/snap/snapcraft.yaml#L253

**Comment by kylestein-arm** [comment]
Removed patch file and updated sha for edge-proxy

**Pull request merged**